querqy-core/src/main/java/querqy/rewrite/RewriteChain.java

```diff
@@ -45,41 +45,95 @@ public RewriteChain(final List<RewriterFactory> factories) {
         });
     }
 
-    public ExpandedQuery rewrite(final ExpandedQuery query,
-                                 final SearchEngineRequestAdapter searchEngineRequestAdapter) {
-      
-        RewritingOutput rewrittenQuery = new RewritingOutput(query);
+    public RewriteChainOutput rewrite(final ExpandedQuery query,
+                                      final SearchEngineRequestAdapter searchEngineRequestAdapter) {
 
-        final Optional<InfoLoggingContext> loggingContext = searchEngineRequestAdapter.getInfoLoggingContext();
+        final RewritingExecutor executor = new RewritingExecutor(factories, searchEngineRequestAdapter, query);
+        return executor.rewrite();
+    }
+
+    @Deprecated
+    public List<RewriterFactory> getRewriterFactories() {
```

Comment:
As far as I can see, this method is not used (same like the other Deprecated getFactory(1x)). Can we just remove it?